src/main/java/io/jenkins/pluginhealth/scoring/probes/CompoundUpdateCenterProbes.java

```diff
@@ -0,0 +1,53 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.util.List;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = CompoundUpdateCenterProbes.ORDER)
+public class CompoundUpdateCenterProbes extends Probe implements UpdateCenterProbe {
```

Comment:
This is `Container (aka composite)` based on the composite design pattern. Ideally, it should:
 >  download the update-center.json file and parse it. Then, it would call other classes which would take the Plugin and the UpdateCenter classes in a method as argument.

But my implementation of `readUpdateCenter` below is wrong and doesn't make any difference because each probe reads the update center from the scratch by itself. I couldn't make sense of how I could pass the `UpdateCenter` classes in this method as an argument.

Reply:
The `doApply(Plugin)` would download and parse the file and then call another method on each subprobes with both the `UpdateCenter` and the `Plugin` objects.